modules/ip/datapath/ip_fragment.c

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2025 Anthony Harivel
+
+#include <gr_datapath.h>
+#include <gr_graph.h>
+#include <gr_iface.h>
+#include <gr_ip4_datapath.h>
+#include <gr_log.h>
+#include <gr_mbuf.h>
+#include <gr_trace.h>
+
+#include <rte_byteorder.h>
+#include <rte_ip.h>
+#include <rte_mbuf.h>
+
+#include <stdint.h>
+
+enum {
+	IP_OUTPUT = 0,
+	NO_MBUF,
+	ERROR,
+	EDGE_COUNT,
+};
+
+static uint16_t
+ip_fragment_process(struct rte_graph *graph, struct rte_node *node, void **objs, uint16_t nb_objs) {
+	struct rte_mbuf *mbuf, *frag_mbuf;
+	struct rte_ipv4_hdr *ip, *frag_ip;
+	const struct iface *iface;
+	uint16_t ip_hdr_len, mtu;
+	uint16_t frag_size, frag_data_len;
+	uint16_t data_len, offset;
+	uint16_t num_frags, i;
+	uint16_t frag_id;
+	rte_edge_t edge;
+	void *payload;
+
+	for (uint16_t j = 0; j < nb_objs; j++) {
+		mbuf = objs[j];
+		ip = rte_pktmbuf_mtod(mbuf, struct rte_ipv4_hdr *);
+
+		// Check if packet is already a fragment - if so, just pass it through
+		if ((ip->fragment_offset
+		     & rte_cpu_to_be_16(RTE_IPV4_HDR_MF_FLAG | RTE_IPV4_HDR_OFFSET_MASK))
+		    != 0) {
+			// This is already a fragment, drop it
+			edge = ERROR;
```

Comment:
Would be interesting to have more precise error nodes, e.g.: `ALREADY_FRAGMENTED` ?